cmd/heimdalld/main.go

```diff
@@ -138,6 +184,254 @@ func exportAppStateAndTMValidators(logger log.Logger, db dbm.DB, storeTracer io.
+				return startStandAlone(ctx, appCreator)
+			}
+
+			ctx.Logger.Info("starting ABCI with Tendermint")
+
+			startRestServer, _ := cmd.Flags().GetBool(helper.RestServerFlag)
+			startBridge, _ := cmd.Flags().GetBool(helper.BridgeFlag)
+
+			_, err := startInProcess(ctx, appCreator, cdc, startRestServer, startBridge)
+			return err
+		},
+		PreRun: func(cmd *cobra.Command, args []string) {
+			// bridge binding
+			if err := viper.BindPFlag("all", cmd.Flags().Lookup("all")); err != nil {
+				logger.Error("GetStartCmd | BindPFlag | all", "Error", err)
+			}
+
+			if err := viper.BindPFlag("only", cmd.Flags().Lookup("only")); err != nil {
+				logger.Error("GetStartCmd | BindPFlag | only", "Error", err)
+			}
+		},
+	}
+
+	cmd.Flags().Bool(
+		helper.RestServerFlag,
+		false,
+		"Start rest server",
+	)
+
+	cmd.Flags().Bool(
+		helper.BridgeFlag,
+		false,
+		"Start bridge service",
+	)
+
+	cmd.PersistentFlags().String(helper.LogLevel, ctx.Config.LogLevel, "Log level")
+	if err := viper.BindPFlag(helper.LogLevel, cmd.PersistentFlags().Lookup(helper.LogLevel)); err != nil {
+		logger.Error("main | BindPFlag | helper.LogLevel", "Error", err)
+	}
+	// bridge flags
+	cmd.Flags().Bool("all", false, "start all bridge services")
+	cmd.Flags().StringSlice("only", []string{}, "comma separated bridge services to start")
+
+	// rest server flags
+	cmd.Flags().String(client.FlagListenAddr, "tcp://0.0.0.0:1317", "The address for the server to listen on")
+	cmd.Flags().Bool(client.FlagTrustNode, true, "Trust connected full node (don't verify proofs for responses)")
+	cmd.Flags().String(client.FlagChainID, "", "The chain ID to connect to")
+	cmd.Flags().String(client.FlagNode, helper.DefaultTendermintNode, "Address of the node to connect to")
+	cmd.Flags().Int(client.FlagMaxOpenConnections, 1000, "The number of maximum open connections")
+
+	// core flags for the ABCI application
+	cmd.Flags().Bool(flagWithTendermint, true, "Run abci app embedded in-process with tendermint")
+	cmd.Flags().String(flagAddress, "tcp://0.0.0.0:26658", "Listen address")
+	cmd.Flags().String(flagTraceStore, "", "Enable KVStore tracing to an output file")
+	cmd.Flags().String(flagPruning, "syncable", "Pruning strategy: syncable, nothing, everything")
+	cmd.Flags().String(
+		FlagMinGasPrices, "",
+		"Minimum gas prices to accept for transactions; Any fee in a tx must meet this minimum (e.g. 0.01photino;0.0001stake)",
+	)
+	cmd.Flags().Uint64(FlagHaltHeight, 0, "Height at which to gracefully halt the chain and shutdown the node")
+	cmd.Flags().Uint64(FlagHaltTime, 0, "Minimum block time (in Unix seconds) at which to gracefully halt the chain and shutdown the node")
+	cmd.Flags().String(flagCPUProfile, "", "Enable CPU profiling and write to the provided file")
+
+	// add support for all Tendermint-specific command line options
+	tcmd.AddNodeFlags(cmd)
+	return cmd
+}
+
+func startStandAlone(ctx *server.Context, appCreator server.AppCreator) error {
```

Comment:
What is the difference between `startStandAlone` and `startInProcess`?

Reply:
This was moved from cosmos SDK, and it will start ABCI with or without Tendermint depending on the flag **with-tendermint**.

Reply:
This is a copy paste from cosmos right? You did not change anything.

Reply:
Yeah, but we're just using the startInProcess func, which means that we can potentially delete this one without any consequences...

Reply:
Ok, lets delete it then.